oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java

```diff
@@ -1393,7 +1393,7 @@ private void updateCoords(String cluster, BundleJob bundle,
 
             //calculate next start time based on delay.
             endTime = (delay == null) ? endTime
-                    : EntityUtil.getNextStartTime(coord.getStartTime(), delay, EntityUtil.getTimeZone(entity), endTime);
+                    : EntityUtil.getNextInstanceTimeWithDelay(endTime, delay, EntityUtil.getTimeZone(entity));
```

Comment:
Thanks for clarifying @sandeepSamudrala. Can you please add UTs for the new method? I understand, ITs may be hard for the whole scenario.